q2_types/feature_data/_format.py

```diff
@@ -221,22 +220,102 @@ class PairedDNASequencesDirectoryFormat(model.DirectoryFormat):
 
 
 class AlignedDNAFASTAFormat(model.TextFileFormat):
-    def sniff(self):
-        filepath = str(self)
-        sniffer = skbio.io.io_registry.get_sniffer('fasta')
-        if sniffer(filepath)[0]:
-            generator = skbio.io.read(filepath, constructor=skbio.DNA,
-                                      format='fasta', verify=False)
+    def _validate_lines(self, max_lines):
```

Comment:
Is there any way we can DRY this up a bit? Can the validation logic be a bit more shared between the formats?

Reply:
Probably. I'm messing with that and a couple of other things right now.

Reply:
You rock, thanks!